addons/pos_coupon/static/src/js/coupon.js

```diff
@@ -274,7 +274,14 @@ odoo.define('pos_coupon.pos', function (require) {
         },
         init_from_JSON: function (json) {
             _order_super.init_from_JSON.apply(this, arguments);
-            this.bookedCouponCodes = json.bookedCouponCodes;
+            this.bookedCouponCodes = {};
+            const filteredLines = json.lines.filter(line => line[2].coupon_id);
+            const couponsCode = filteredLines.map(line => [line[2].coupon_id, line[2].program_id[0]]);
+            couponsCode.forEach((coupon) =>{
+                const [code, program_id] = coupon;
+                this.bookedCouponCodes[code[1]] = new CouponCode(code[1], code[0], program_id);
```

Comment:
we need to do the same also for `this.activePromoProgramIds`

Reply:
I don't think there is this problem with the PromoPrograms. Or maybe I didn't test it right